tensorflow/go/session.go

```diff
@@ -65,6 +65,53 @@ func NewSession(graph *Graph, options *SessionOptions) (*Session, error) {
 	return s, nil
 }
 
+type Device struct {
+	Name, Type  string
+	MemoryLimit int64
+}
+
+// Return list of devices associated with a Session
+func (s *Session) ListDevices() ([]Device, error) {
+	var devices []Device
+
+	status := newStatus()
+	devices_list := C.TF_SessionListDevices(s.c, status.c)
+	if err := status.Err(); err != nil {
+		return nil, fmt.Errorf("TF_SessionListDevices() failed: %v", err)
+	}
+	defer C.TF_DeleteDeviceList(devices_list)
+
+	for i := 0; i < int(C.TF_DeviceListCount(devices_list)); i++ {
+		status = newStatus()
```

Comment:
Do we need to create a `newStatus()` multiple times?Seems like we can re-use the one `status` created in line 77?

Reply:
Done, thanks. I tried to find some info on how safe is to re-use statuses and I found nothing. Other places in code does so though.